pkg/apis/core/validation/validation.go

```diff
@@ -1816,7 +1816,6 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 		allErrs = append(allErrs, ValidateImmutableField(newPvc.Spec.VolumeMode, oldPvc.Spec.VolumeMode, field.NewPath("volumeMode"))...)
 	}
 
-	newPvc.Status = oldPvc.Status
```

Comment:
> So it is. That's a bug. That should happen in the strategy PrepareForCreate/PrepareForUpdate methods, not in validation.

This is already done here:
https://github.com/kubernetes/kubernetes/blob/6c5f0799bdd65ca99394017cb6db54bcc1464961/pkg/registry/core/persistentvolumeclaim/strategy.go#L75

So we can just remove it here.